### DISCUSSION TITLE
K3s invalid capacity 0 on image filesystem. cri_stats_provider.go:455] "Failed to get the info of the filesystem with mountpoint

### DISCUSSION THREAD

**Discussion by madeeldevops** [k3s-io/k3s#8698] [Q&A]
Yes, as the error says, it takes a minute for the stats to be collected. `cri_stats_provider` pulls filesystem utilization from cadvisor, which takes a bit to initialize and collect data before it can be queried.  If you look at the complete logs, you'll see another error preceding it:

```
cri_stats_provider.go:375] Failed to get the info of the filesystem with mountpoint "/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs": unable to find data in memory cache.
```

This should be harmless and clear within a few seconds after the node starts. Are you seeing otherwise?

_Originally posted by @brandond in https://github.com/k3s-io/k3s/issues/1857#issuecomment-637694918_

**Comment by madeeldevops** [comment]
Yes, you are correct but somehow i get different error, i have restart node as well as k3s-agent on my worker node but still.
`Oct 19 04:31:38 nodek2 k3s[41185]: E1019 04:31:38.227900   41185 cri_stats_provider.go:455] "Failed to get the info of the filesystem with mountpoint" err="failed to get device for dir \"/var/lib/lxd/images\": stat failed on /var/lib/lxd/images with error: no such file or directory" mountpoint="/var/lib/lxd/images"`

And just like you said it just warning i ignored but when i was following k3s logs i get these when i was trying to create a pod
`Oct 19 04:36:38 nodek2 k3s[41185]: E1019 04:36:38.086264   41185 kubelet.go:1396] "Image garbage collection failed multiple times in a row" err="invalid capacity 0 on image filesystem"
Oct 19 04:36:38 nodek2 k3s[41185]: E1019 04:36:38.403078   41185 cri_stats_provider.go:455] "Failed to get the info of the filesystem with mountpoint" err="failed to get device for dir \"/var/lib/lxd/images\": stat failed on /var/lib/lxd/images with error: no such file or directory" mountpoint="/var/lib/lxd/images"`

**Comment by brandond** [comment]
Are you running k3s in a lxd container? If so, why? I'm not sure that's expected to work.

**Reply by madeeldevops** [reply]
LXC VM not container. I am new to this, so i was trying a homelab on my linux machine.

**Comment by madeeldevops** [comment]
Yes, I have followed [this](https://github.com/automaticserver/lxe/blob/master/doc/examples/multi-node-cluster-with-k3s.md). Everything works. The container is created and then shutdown after 30 or so.

**Reply by brandond** [reply]
I'm confused, does everything work, or is it not working?

**Reply by madeeldevops** [reply]
It doesn't work. I will post the issue to the maintainer, and will get back here.

**Comment by brandond** [comment]
Ok. That's not something we've tested or would know how to support, I would suggest going to the maintainers of the project whose documentation you are following to ask for help.

**Comment by kamikaze** [comment]
getting the same issue with both containerd and docker containerd.io on a fresh Ubuntu Mantic installation

**Reply by brandond** [reply]
Yes, as discussed above, it is normal to see this on startup. Is there a problem?

**Reply by kamikaze** [reply]
I see it as a red error in Lens and no pods are being launched. It was when I did install k3s without flannel for the first time. Even installing cilium didn't help. But once I re-installed with flannel - stuff just worked and then I was able to swap it with cilium.

**Reply by brandond** [reply]
This error has nothing to do with the CNI, I suspect that something else is going on.
